src/ariel-os-debug-log/src/lib.rs

```diff
@@ -55,7 +55,11 @@ mod log_macros {
     macro_rules! trace {
         ($($arg:tt)*) => {{
             use $crate::defmt::hidden::defmt;
-            defmt::trace!($($arg)*);
+            if true {
+                defmt::trace!($($arg)*);
+            } else {
+                drop(format_args!($($arg)*));
+            }
```

Comment:
If I'm not mistaken, I think you can achieve the same thing by simply adding an `if false`, which makes this easier to document:

```suggestion
            defmt::trace!($($arg)*);
            if false {
                drop(format_args!($($arg)*));
            }
```

Reply:
Can you also add a `NOTE` comment next to the one on this module to document these weird `if`s? The only documentation currently is in the commit message, which makes this a bit difficult to discover.

Reply:
We can't have only an if-false branch because there are a few cases in which $arg is moving.

The comment is now in there in a fixup.